### ISSUE TITLE
Please use the public version of Microsoft.Rest.ClientRuntime so you don't break other modules

### ISSUE THREAD

**Issue by rusanu** [Azure/azure-powershell#11198]
The Az library ships with its own private version of Microsoft.Rest.ClientRuntime. This is different from the one in the public NuGet feed, even for the same version and public key token...
If the Az module is loaded first into a PowerShell session, any other module that uses an auto-rest generated client will break because the the modules are incompatible:

```
Method not found: '!!0 Microsoft.Rest.Serialization.SafeJsonConvert.DeserializeObject
```

The public Microsoft.Rest.ClientRuntime manifest is binding to an old Newtonsoft JSON:
```
.assembly extern Newtonsoft.Json
{
  .publickeytoken = (30 AD 4F E6 B2 A6 AE ED )                         // 0.O.....
  .ver 6:0:0:0
}
```
the one you guys ship privately uses v. 10:
```
.assembly extern Newtonsoft.Json
{
  .publickeytoken = (30 AD 4F E6 B2 A6 AE ED )                         // 0.O.....
  .ver 10:0:0:0
}
```
this makes the `[Newtonsoft.Json]Newtonsoft.Json.JsonSerializerSettings` parameter passed to ` !!0 Microsoft.Rest.Serialization.SafeJsonConvert::DeserializeObject<!!0>(string,                                                                                                class [Newtonsoft.Json]Newtonsoft.Json.JsonSerializerSettings)` a different type and triggers the runtime error.

If you  want to change the public Microsoft.Rest.ClientRuntime to bind Newtonsoft.JSON at v.10, by al means, do. But do it in the public feed of *that* module, so that any other dependency can update and refresh. Don't ship it as a private DLL in your module and break modules that just happens to load after you in the session...

**Comment by markcowl** [comment]
I am curious what your actual end issue is, because we are using the public version of Microsoft.Rest.ClientRuntime.

If you look at the package here: https://www.nuget.org/packages/Microsoft.Rest.ClientRuntime/ it contains multiple dlls for different targets with multiple Newtonsoft.json dependencies.  Since Az is a NetStandard module, it uses the NetStandard version of this package, which has the Newtonsoft.Json 10 dependency.

If you use the .Net4.5.1 target in your project, then you will get Netwonsoft.Json 6, which is likely the source opf the discrepancy you are seeing.

If you cna provide some context around the issue you are seeing, we cna likely help with a solution.

**Comment by rusanu** [comment]
Thanks for looking at this. I did not realize Az module is .Net Standard and Microsoft.Rest.ClientRuntime binds to different Newtonsoft.Json on different platforms.
The problem I'm facing is this: I have a .Net Framework PowerShell module that contains an auto-rest  C# client. This requires Microsoft.Rest.ClientRuntime, and it uses the .Net Framework one. Clients have complained about the module triggering sometime this error:

```
 Method not found: '!!0 Microsoft.Rest.Serialization.SafeJsonConvert.DeserializeObject(System.String, Newtonsoft.Json.JsonConverter[])'.
```

The error would fix itself if user opened a new PowerShell session, and eventually I tracked this to whether the Az module was loaded first.  When my module loads later, the Microsoft.Rest.ClientRuntime is already loaded. Whenever an auto-rest generated proxy is invoked in my code, when it reaches serialization, it will pass a `JsonSerializerSettings` from my module, which causes the runtime error above.
When I created the issue I thought that Az module ships a private Microsoft.Rest.ClientRuntime version. But if I understand correctly, within a PowerShell session all the auto-rest, and other code that uses Microsoft.Rest.ClientRuntime, all would have to somehow agree to use the same Newtonsoft.Json version (since adding a .config binding redirect to powershell.exe is not feasible for a Powershell module...)
